robyn/__init__.py

```diff
@@ -427,33 +438,70 @@ class SubRouter(Robyn):
     def __init__(self, file_object: str, prefix: str = "", config: Config = Config()) -> None:
         super().__init__(file_object, config)
         self.prefix = prefix
+        self.sub_router = []
+        self.temp_routes = []  # Temporary storage for routes
```

Comment:
Could you explain the idea behind these memebers?

Reply:
Hi I'm sorry there is some issue before when register route. I already fix this. Please check again. I also add some feature to handle multiple auth handler like I said in comment before.

Reply:
Also i add some pytest to ensure the function is working

Reply:
The idea is we save all the nested subrouter and route before the main robyn call include router. Like what I explain in this comment https://github.com/sparckles/Robyn/pull/889#discussion_r1676772575